WhatsApp/app/src/main/java/com/example/whatsapp/ui/fragments/settings/notifications/NotificationsFragment.kt

```diff
@@ -1,19 +1,250 @@
 package com.example.whatsapp.ui.fragments.settings.notifications
 
+import android.app.AlertDialog
+import android.app.Notification
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.content.Context
+import android.content.Context.NOTIFICATION_SERVICE
+import android.graphics.Color
+import android.os.Build
 import android.os.Bundle
-import androidx.fragment.app.Fragment
+import android.view.HapticFeedbackConstants
+import android.view.LayoutInflater
+import android.view.View
+import androidx.core.content.ContextCompat.getSystemService
 import androidx.navigation.NavController
 import com.example.whatsapp.R
+import com.example.whatsapp.base.BaseFragment
+import com.example.whatsapp.databinding.FragmentNotificationsBinding
 import dagger.hilt.android.AndroidEntryPoint
 
+
 @AndroidEntryPoint
-class NotificationsFragment : Fragment(R.layout.fragment_notifications) {
+class NotificationsFragment : BaseFragment<FragmentNotificationsBinding>(){
     private lateinit var navController: NavController
 
+
+    override fun inflateBinding(layoutInflater: LayoutInflater): FragmentNotificationsBinding {
+        return FragmentNotificationsBinding.inflate(layoutInflater)
+    }
+
     override fun onCreate(savedInstanceState: Bundle?) {
         setHasOptionsMenu(true)
         super.onCreate(savedInstanceState)
     }
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        observeUiInteractions()
 
+    }
+    /**
+     * Vibrates the device. Used for providing feedback when the user performs an action.
+     */
+    fun vibrate(view: View) {
+        view.performHapticFeedback(HapticFeedbackConstants.LONG_PRESS)
+    }
+    private fun defaultLed(){
+        val notificationManager= context?.getSystemService(NOTIFICATION_SERVICE) as NotificationManager
 
+        val notification_Channel_ID = "my_channel_id_01"
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            val notificationChannel = NotificationChannel(
+                notification_Channel_ID,
+                "My Notifications",
+                NotificationManager.IMPORTANCE_HIGH
+            )
+
+            // Configure the notification channel.
+
+            notificationChannel.description = "Channel description"
+            notificationChannel.enableLights(true)
+            notificationChannel.lightColor = Color.RED
+            notificationChannel.vibrationPattern = longArrayOf(0, 1000, 500, 1000)
+            notificationChannel.enableVibration(true)
+            notificationManager.createNotificationChannel(notificationChannel)
+        }
+
+    }
+    private fun observeUiInteractions() {
+        //Message Layouts
+        binding.vibrateLayout.setOnClickListener {
```

Comment:
this is fine, although i'd reduce repeated code and also the size of thie entire function. I'd do something like : 

binding.vibrateLayout.setonclicklistener { 
     showOptionsDialog()
}

private fun showOptionsDialog(title : String, choices : List<String or @ResStringId>, ) { 
```suggestion
        binding.vibrateLayout.setOnClickListener {
```
            val builder = AlertDialog.Builder(context)
            builder.setTitle("Vibrate")
            val vibrates = arrayOf("Off", "Default", "Short", "Long")
            val checkedItem = 1
            builder.setSingleChoiceItems(vibrates, checkedItem) { dialog, which ->
                val position = (dialog as AlertDialog).listView.checkedItemPosition
                when(position){
                0 -> //
                1 -> //
                }

                dialog.dismiss()
            }
            val dialog = builder.create()
            dialog.show()
} 

basically make this function reusable

Reply:
We should avoid using hard coded text, also this can be simplified to something like below:

`      
        val vibrates = arrayOf("Off", "Default", "Short", "Long")
        val checkedItem = 1
        AlertDialog.Builder(requireContext())
            .setTitle("Vibrate")
            .setSingleChoiceItems(vibrates, checkedItem) { dialog, position ->
                when (position) {
                    0 -> {}
                    1 -> {}
                }
                dialog.dismiss()
            }
            .create()
            .show()
            `
            
Note: Also look at how you were trying to find the position of the selected item in the list view in addition to the other changes. 

Also, I think it's a good shout if instead of having all of the application defaults settings in random functions in the app we should store them in some configuration file which is solely responsible for housing all of our application default values.